### ISSUE TITLE
PR creation does not trigger workflow

### ISSUE THREAD

**Issue by andy-maier** [actions/github-script#484]
**Describe the bug**

I want to automate the creation of a PR to release a Python package with a "release_pr" workflow triggered by the push of the underlying branch, as follows (full workflow: [release_pr.yml](https://github.com/zhmcclient/python-zhmcclient/blob/master/.github/workflows/release_pr.yml)):

```
name: release_pr

on:
  push:  # When pushing a release branch
    branches:
    # Release branch formats:
    # release_M.N.U - targets a default branch (master for M.N.0, stable_M.N otherwise)
    # release_M.N.0_BRANCH - targets the specified branch
    - release_*

jobs:
  release_pr:
    name: Create the release PR
    runs-on: ubuntu-latest
    steps:

    . . .

    - name: Create PR on Github for the pushed release branch
      uses: actions/github-script@v7
      with:
        script: |
          github.rest.pulls.create({
            owner: context.repo.owner,
            repo: context.repo.repo,
            title: "Release ${{ steps.set-version.outputs.result }}",
            body: "Merging this PR will trigger the publishing to PyPI.",
            head: "${{ github.ref_name }}",
            base: "${{ steps.set-target-branch.outputs.result }}",
          });
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

The PR gets created successfully by this workflow, e.g.: https://github.com/zhmcclient/python-zhmcclient/pull/1641

There is a second workflow "test" that is set to trigger when a PR is created. However, that second workflow is not actually triggered when the PR is created using actions/github-script (but it is triggered when manually creating a PR).

The trigger section of the "test" workflow is as follows (full workflow: [test.yml](https://github.com/zhmcclient/python-zhmcclient/blob/master/.github/workflows/test.yml))

```
name: test

on:
  schedule:
    # The schedule event always (and only) runs on the master branch.
    - # cron (in UTC): minute hour day_of_month month day_of_week
      cron: '00 22 * * SAT'
  pull_request:  # When creating a PR targeting these branches
    branches:
    - master
    - stable_*
  push:  # When merging a PR targeting these branches (direct push is disabled)
    branches:
    - master
    - stable_*

jobs:
  . . .
```

Since the created release PR targets either the "master" or "stable_M.N" branch, the "pull_request" trigger should trigger when a PR is created through the actions/github-script code (just as it does when such a PR is created manually), but it is not actually triggered.

Do you have an idea why the "test" workflow it is not triggered through the actions/github-script code?

What is different in the way the actions/github-script code creates the PR compared to creating the PR manually, that would influence the subsequent triggering of actions?

**To Reproduce**
Steps to reproduce the behavior (requires push permissions to the repo):
1. Clone the repo https://github.com/zhmcclient/python-zhmcclient locally 
2. Create a virtual Python environment using Python 3.8 or higher and have that env active
3. Go to the main directory of the local repo clone
3. Run "make develop" - this will install necessary Python packages
4. Run "VERSION=1.18.0 make release_branch" - this will create a release branch and push that branch, which triggers the "release_pr" workflow.
5. Look at the "Actions" tab to verify that the "test" workflow was not run.

**Expected behavior**

PR creation through the actions/github-script code should have the same Actions trigger behavior as creating a PR manually through the Github GUI.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: MacOS 14.6.1
 - Browser: Firefox
 
**Smartphone (please complete the following information):**
N/A

**Additional context**
N/A

**Comment by joshmgross** [comment]
Events triggered by the `GITHUB_TOKEN` do not trigger other Actions workflows, with the exception of `workflow_dispatch` and `repository_dispatch` workflows.

> When you use the repository's `GITHUB_TOKEN` to perform tasks, events triggered by the `GITHUB_TOKEN`, with the exception of `workflow_dispatch` and `repository_dispatch`, will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs.

https://docs.github.com/actions/security-for-github-actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow


So your `release_pr` workflow will need to use a PAT or GitHub App token to create that pull request.

**Comment by andy-maier** [comment]
I have tried both a GitHub App token and a PAT, but no success. In all cases, the PR got created, but the subsequent test workflow was not triggered.

On the GitHub App, I'm not sure I did that right because I registered a dummy app that does not actually exist, just to get an ID and private key. The app had permissions to all repos in the org and r/w to actions (I probably should have added r/w access to PRs and/or content).

With the PAT, I do think I did that right. I created a fine grained PAT with permissions for all repos in the org, and R/W access to actions, content and pull requests. I then put it into an org level secret and used that secret as the GITHUB_TOKEN in the workflow step that used "actions/github-script" for creating the PR. See https://github.com/zhmcclient/python-zhmcclient/pull/1645/files for the workflow.

Is there anything else I may be missing?

**Comment by joshmgross** [comment]
Looking at the latest release PR, it's still being created as the Actions bot

https://github.com/zhmcclient/python-zhmcclient/pull/1647

```yaml
    - name: Create PR on Github for the pushed release branch
      uses: actions/github-script@v7
      with:
        script: |
          github.rest.pulls.create({
            owner: context.repo.owner,
            repo: context.repo.repo,
            title: "Release ${{ steps.set-version.outputs.result }}",
            body: "Merging this PR will trigger the publishing to PyPI.",
            head: "${{ github.ref_name }}",
            base: "${{ steps.set-target-branch.outputs.result }}",
          });
      env:
        GITHUB_TOKEN: ${{ secrets.PAT_ZHMCCLIENT_CREATE_PR }}
```

I believe you need to use the `github-token` action input, rather than setting the `GITHUB_TOKEN` environment variable.


https://github.com/actions/github-script/blob/35b1cdd1b2c1fc704b1cd9758d10f67e833fcb02/action.yml#L11-L14

https://github.com/actions/github-script?tab=readme-ov-file#using-a-separate-github-token


So that would look something like:

```yaml
    - name: Create PR on Github for the pushed release branch
      uses: actions/github-script@v7
      with:
        github-token: ${{ secrets.PAT_ZHMCCLIENT_CREATE_PR }}
        script: |
          github.rest.pulls.create({
            owner: context.repo.owner,
            repo: context.repo.repo,
            title: "Release ${{ steps.set-version.outputs.result }}",
            body: "Merging this PR will trigger the publishing to PyPI.",
            head: "${{ github.ref_name }}",
            base: "${{ steps.set-target-branch.outputs.result }}",
          });
```

**Comment by andy-maier** [comment]
With this change, the actions/github-script now gets a HTTP status 403 with message "Resource not accessible by personal access token", see https://github.com/zhmcclient/python-zhmcclient/actions/runs/10787652817/job/29916779256

I then switched back to use the dummy app and actions/create-github-app-token to create an app tolen for it. I added several repo-level permissions to the dummy app, but that still resulted in the same error.

I will give up on the idea to automate the PR creation; it would have been only a small improvement for the person that performs the release.

Thanks for your help anyway!
